### PR TITLE
[pickers] Fix to avoid selecting sections on mobile picker field

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #8145

### CODE REVIEW

File: packages/x-date-pickers-pro/src/MultiInputDateRangeField/MultiInputDateRangeField.tsx [hunk 1]
```diff
@@ -285,6 +287,11 @@ MultiInputDateRangeField.propTypes = {
       startIndex: PropTypes.number.isRequired,
     }),
   ]),
+  /**
+   * Control whether the input selection is controlled by field.
```
Comment (alexfauquette) [inline packages/x-date-pickers-pro/src/MultiInputDateRangeField/MultiInputDateRangeField.tsx:291]: Not sure
```suggestion
   * Control whether the input selection is controlled by the field.
```

File: packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateRangeField.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ export const useMultiInputDateRangeField = <TDate, TChildProps extends {}>({
   const adapter = useLocalizationContext<TDate>();
 
   const { value: valueProp, defaultValue, format, onChange, disabled, readOnly } = sharedProps;
+  const { shouldControlSelection, ...otherSharedProps } = sharedProps;
```
Comment (alexfauquette) [inline packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateRangeField.ts:40]: Why should it be extracted from `sharedProps` if it is shared between the two inputs?

**Review by alexfauquette — COMMENTED** [review]
Sound nice. I'm just unsure about what `shouldControlSelection={false}` implies.

In usual cases, it's not a problem, because either you don't want input selection at all (in mobile case) or you want to control it (desktop mode)

But I tested it with date range field, and depending on the number of `<input/>` it does not have the same behavior.

[Screencast from 14-03-2023 13:49:55.webm](https://user-images.githubusercontent.com/45398769/225006537-3417bc0c-dfbb-4550-803e-06a036d70d6a.webm)

**Comment by LukasTy** [comment]
@alexfauquette I forgot about the fact, that we have `readOnly` flag, which is always set on `mobile` pickers and it does somewhat make sense to tie the input selection controlling logic with this flag.
It does make for a really small change. 🙈 
What do you think? 🤔

**Comment by alexfauquette** [comment]
Nice, for me it solves the problem 🎉

**Comment by LukasTy** [comment]
Can anyone @mui/x test if this actually fixes the underlying issue on an iOS device?

**Comment by flaviendelangle** [comment]
I'm using Android, I can check tonight on someone's Iphone if needed

**Review by flaviendelangle — APPROVED** [review]
Nice improvement :partying_face:

**Pull request merged**